lib/core/src/chain/bitcoin.rs

```diff
@@ -110,7 +110,7 @@ impl BitcoinChainService for HybridBitcoinChainService {
         match maybe_popped_header {
             Some(popped_header) => {
                 let tip: HeaderNotification = popped_header.try_into()?;
-                self.tip = tip;
+                *self.tip.lock().unwrap() = tip;
```

Comment:
Can we more safely unwrap the lock here and the lines below?

Reply:
I actually think we want to panic here. The only reason the lock() returns error is if one of the holders thread panics. I think we want to propagate that instead of hiding it and keep maintaining an unusable poisoned lock.

Reply:
The only reason for this to panic here is if one of the holder threads panics. I think we want to propagate that rather then maintaining an unusable poisoned lock.

Reply:
Just to understand, this means our `track_new_blocks` thread will panic in most cases and any call from the main thread if `next_unused_address()` is used?